src/prepare_output_secondary.F90

```diff
@@ -330,6 +331,44 @@ subroutine prepare_output_secondary(Ctrl, i, j, MSI_Data, SPixel, Diag, &
            sreal_fill_value, sint_fill_value)
    end do
 
+   !----------------------------------------------------------------------------
+   ! Measurement error (diagonals)
+   !----------------------------------------------------------------------------
+   if (Ctrl%Ind%flags%do_meas_error) then
+      ! Check if a valid Sy is available for output.
+      ! This is a little messy, because of the way SPixel%Sy is allocated and
+      ! deallocated for each SPixel. We need to check:
+      ! * If we have a measurement count (Ny) - this is reset for each pixel,
+      !   and can be zero if the spixel is skipped early on.
+      ! * If Sy array has actually been defined - as Sy is deallocated and
+      !   then reallocated for each pixel, it is possible that, if the first
+      !   pixel is skipped _after_ Ny has been defined, Sy will not be
+      !   allocated.
+      ! * If the Sy array has the correct dimension. If we have missing
+      !   channels and skipped pixels, it is also possible at Ny ends up
+      !   being larger than Sy, at this point in the code.
+      if (SPixel%Ind%Ny .gt. 1 .and. size(SPixel%Sy) .gt. 1 .and. &
+           SPixel%Ind%Ny*SPixel%Ind%Ny .le. size(SPixel%Sy)) then
+         do k=1,SPixel%Ind%Ny
+            !write(*,"(e9.2)", advance="no") sqrt(SPixel%Sy(k,k))
+            call prepare_short_packed_float( &
+                 sqrt(SPixel%Sy(k,k)), output_data%Sy(i,j,k), &
```

Comment:
This doesn't map channels between SPixel and Ctrl and so is outputting in the wrong place. I'll work out a fix momentarily.